models/amity.py

```diff
@@ -0,0 +1,338 @@
+                        else:
+                            cls.room_allocations[room_name] = [full_name]
+                else:
+                    print("cannot reallocate staff to a living space")
+        else:
+            print("Room {0} not created!".format(room_name))
+
+    @classmethod
+    def load_people(cls, filename):
+        """ Loads people from the specified file
+
+        Arguments:
+        filename: name of file containing data
+        """
+        if filename:
+            with open(filename, "r") as file:
+                lines = file.readlines()
+                for line in lines:
+                    person_details = line.split()
+                    if len(person_details) == 3:
+                        first_name = person_details[0]
+                        last_name = person_details[1]
+                        person_type = person_details[2]
+                        wants_accomodation = "N"
+                        cls.add_person(first_name=first_name, last_name=last_name, person_type=person_type,
+                                       wants_accomodation=wants_accomodation)
+                    elif len(person_details) == 4:
+                        first_name = person_details[0]
+                        last_name = person_details[1]
+                        person_type = person_details[2]
+                        wants_accomodation = person_details[3]
+                        cls.add_person(first_name=first_name, last_name=last_name, person_type=person_type,
+                                       wants_accomodation=wants_accomodation)
+                    else:
+                        print("could not process provided data!")
+        else:
+            print("Please provide a file!")
+
+    # Helper Method to convert dictionary keys to a list
+    @staticmethod
+    def convert_keys_to_list(allocation_keys):
+        """ Converts a dictionary's keys from type dict_keys to type list
+
+        Arguments:
+        allocation_keys: room_allocation keys
+
+        Return:
+        output_list: list containing the keys
+        """
+        output_list = []
+        for item in allocation_keys:
+            output_list.append(item)
+
+        return output_list
+
+    @classmethod
+    def print_allocations(cls, filename=""):
+        """ Display the currently created rooms and people in them
+
+        Arguments:
+        filename: if a filename is provided then this function prints to the file
+        """
+        if filename:
+            print("\nwriting to file...\n")
+            with open(filename, 'w') as file:
+                rooms = Amity.convert_keys_to_list(cls.room_allocations.keys())
+                for room in rooms:
+                    file.write(room.upper() + "\n")
+                    file.write("-" * 50 + "\n")
+                    people = cls.room_allocations[room]
+                    str_people = ", ".join(people)
+                    file.write(str_people + "\n")
```

Comment:
a good idea is to build up the text first and then call `file.write` only once to write everything to file at a go